kernel/fs/iso9660.inc

```diff
@@ -736,3 +736,1044 @@ cd_compare_name:
         pop     esi eax edi edx
         stc
         ret
+
+;;; New driver
+
+DESCRIPTOR_TYPE_BOOT          = 0x00    ; not used
+DESCRIPTOR_TYPE_PRIMARY       = 0x01
+DESCRIPTOR_TYPE_SUPPLEMENTARY = 0x02
+; DESCRIPTOR_TYPE_PRIMARY for Juliet(UTF-16)
+DESCRIPTOR_TYPE_PARTITION     = 0x02
+DESCRIPTOR_TYPE_TERMANATOR    = 0xff    ; end scan descriptors
+
+struct  ISO9660_VOLUME_DESCRIPTOR
+        type    rb 1
+        magic   rb 5  ; 'CD001'
+        version rb 1
+ends
+
+struct  ISO9660_PRIMARY_DESCRIPTOR  ISO9660_VOLUME_DESCRIPTOR
+                        rb 1
+                        rb 32
+        VolumeName      rb 32 ; ascii or utf-16
+                        rb 8
+        VolumeSpaceSize rb 8
+                        rb 32
+                        rb 4 ; +120
+                        rb 4
+        LBA_Size        rb 4 ; +128
+        PathTableSize   rb 8         ; Do not use it!!!
+        LBA_PathTable   rb 4 ; +140  ; Do not use it!!!
+                        rb 4 ; +144  ; Do not use it!!!
+                        rb 8
+        root_dir_record rb 34  ;ISO9660_DIRECTORY_RECORD
+                        rb 128*4 + 37*3
+        date_created    rb 17
+        date_modiffed   rb 17
+                        rb 17
+        file_struct_ver rb 1 ;always 0x01
+ends
+
+struct  ISO9660_DIRECTORY_RECORD
+        size            rb 1
+                        rb 1
+        lba             rd 1
+                        rd 1
+        data_length     rd 1
+                        rd 1
+        date_time       rb 7
+        flags           rb 1 ; 10b - directory  1b - hidden
+                        rb 2+4
+        name_len        rb 1
+        name            rb 1 ;  rb [name_len]
+ends
+
+; Internal data for every ISO9660 partition.
+struct ISO9660 PARTITION
+       type_encoding    rd 1 ; 0 - ascii 1 - UCS-2
+       lba_size         rd 1 ; default 2048 TODO
+       primary_descr    rd 1 ; sector num
+       root_dir_lba     rd 1 ; lba32
+       root_dir_len     rd 1 ;
+ends
+;-----------------------------------------------------------------------------
```

Comment:
Perhaps we should get rid of these strange separators (`;--------------------------`). I don’t understand on what principle they are arranged.

Reply:
these separators are needed to visually separate functions and code blocks within them.